### PR TITLE
Validate issuer and audience

### PR DESCRIPTION
**Pull request by pegtrifork**

### CODE REVIEW

File: docker-compose/kafka.yaml [hunk 1]
```diff
@@ -185,7 +185,7 @@ services:
       # quarkus.log.level: DEBUG
       QUARKUS_OIDC_ROLES_ROLE_CLAIM_PATH: "roles"
       QUARKUS_OIDC_AUTHENTICATION_VERIFY_ACCESS_TOKEN: "true"
-      QUARKUS_OIDC_TOKEN_AUDIENCE: any # should be "schema-registry,kafka" but our keycloak does not set 'aud' on tokens
+      QUARKUS_OIDC_TOKEN_AUDIENCE: schema-registry,kafka
```
Comment (cthtrifork) [inline docker-compose/kafka.yaml:188]: According to your insights, should it not be
```suggestion
      QUARKUS_OIDC_TOKEN_AUDIENCE: schema-registry
```

File: config/kafka/server.properties [hunk 2]
```diff
@@ -7,24 +7,29 @@ listener.security.protocol.map=auth-listener-docker-internal:SASL_PLAINTEXT,auth
 sasl.mechanism.inter.broker.protocol=OAUTHBEARER
 
 # Internal docker oauth2 client, which is also used for inter-broker communication.
+# "oauth.client.id" is the valid audience
 listener.name.auth-listener-docker-internal.sasl.enabled.mechanisms=OAUTHBEARER
 listener.name.auth-listener-docker-internal.oauthbearer.sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
-    oauth.client.id="kafka-admin" \
-    oauth.client.secret="kafka-admin-secret" \
```
Comment (cthtrifork) [inline config/kafka/server.properties:13]: why this change

**Comment by cthtrifork** [comment]
Shouldn't we add it for opensearchconfig.yaml as well? and update schema-registry to verify audience.

File: config/kafka/server.properties (hunk 2 above)
Comment (pegtrifork) [reply config/kafka/server.properties:13]: Because oauth.client.id is used for both the valid audience and for getting a token for inter-broker communication, and I would like the audience to be "kafka". Therefore I added a client in keycloak with the client-id kafka, and made this change

**Comment by pegtrifork** [comment]
> Shouldn't we add it for opensearchconfig.yaml as well? and update schema-registry to verify audience.

I think you might be looking at and old/wrong diff. Both opensearch and schema-registry validates both issuer and audience

File: config/kafka/server.properties (hunk 2 above)
Comment (cthtrifork) [reply config/kafka/server.properties:13]: I have a really bad time reading `config/keycloak/local-development.json` file, but it seems strange that there is a kafka-admin user which is not being used internally by Kafka.

Redpanda, however, uses the `kafka-admin`.
So I think the naming is confusing

Comment (pegtrifork) [reply config/kafka/server.properties:13]: I changed it so now redpanda uses the `redpanda` client credentials, and kafka setup also uses the new `kafka` client. Which means that the `kafka-admin` is not used internally anymore and could be removed. But I would like to keep it to not break any potential clients using it.

Comment (cthtrifork) [reply config/kafka/server.properties:13]: I vote to remove it, as clients outside this repo should not be using it?

**Review by cthtrifork — APPROVED** [review]
